### DISCUSSION TITLE
High(er) speed power measurment

### DISCUSSION THREAD

**Discussion by NullByteZero** [arendst/Tasmota#22276] [Q&A] [answered]
Hi! Does Tasmota support any power measurement boards with high speed (few Hz) power measurement? Or is there an option to speedup the measurement interval in the software? 

I've found a project on github which does exactly that by modifying Shelly 2.5 firmware.
https://github.com/josemariaaraujo/Shelly-Power-Diverter/tree/main

If this isn't already implemented in tasmota is there a reason behind it? If not I would happily implement this feature and create a PR. I've seen quite a lot of "Solar Diverter" type projects which work best at high speeds. Maybe Tasmota could have its own native implementation :)

**Comment by sfromis** [comment]
What Tasmota does is energy consumption monitoring, not suitable for process control where the purpose is **instead** very quick reactions to changes of current/voltage.

**Reply by NullByteZero** [reply]
Why do you think it isn't "suitable for process control"? Would it hurt if we make it suitable for even for this purpose? I get why you can't add this feature to all of the supported devices. But there are devices out there which use hardware ICs which could make this possible.

**Reply by sfromis** [reply, marked as answer]
I just told you why, as you do not get very quick millisecond-level reactions to changes. That's just not how the energy consumption feature works, where the aim is to count watts, with voltage/current/frequency data only being additional info happening to be available.

And my comment was about how it currently works, not about how feasible (or not) changes to support such expectations would be.

**Comment by nikito7** [comment]
Create a driver 

https://github.com/phpcoder/tasmota-custom-integration/
